config/confighttp/confighttp.go

```diff
@@ -148,6 +148,11 @@ type HTTPServerSettings struct {
 	// CORS needs to be enabled first by providing a non-empty list in CorsOrigins
 	// A wildcard (*) can be used to match any header.
 	CorsHeaders []string `mapstructure:"cors_allowed_headers"`
+
+	// CorsMaxAge allows browsers to cache CORS preflight responses. It sets
+	// the Access-Control-Max-Age header. See github.com/rs/cors
+	// CORS needs to be enabled first by providing a non-empty list in CorsOrigins.
+	CorsMaxAge int `mapstructure:"cors_max_age"`
```

Comment:
Is this the time when we should pull the trigger and create a "CorsSettings"? Seems that we are adding more and more things related to "cors". I can see that in the future more things may be added here.

This will be a breaking change but I think will improve things:

```

type HTTPServerSettings struct {
  // ... 
	// CorsSettings struct exposes Cors configuration.
	CorsSettings *CorsSettings `mapstructure:"cors, omitempty"`
}

CorsSettings {
	// Origins are the allowed CORS origins for HTTP/JSON requests to grpc-gateway adapter
	// for the OTLP receiver. See github.com/rs/cors
	// An empty list means that CORS is not enabled at all. A wildcard (*) can be
	// used to match any origin or one or more characters of an origin.
	Origins []string `mapstructure:"allowed_origins"`
	// CorsHeaders are the allowed CORS headers for HTTP/JSON requests to grpc-gateway adapter
	// for the OTLP receiver. See github.com/rs/cors
	// CORS needs to be enabled first by providing a non-empty list in CorsOrigins
	// A wildcard (*) can be used to match any header.
	Headers []string `mapstructure:"allowed_headers"`

	// MaxAge allows browsers to cache CORS preflight responses. It sets
	// the Access-Control-Max-Age header. See github.com/rs/cors
	// CORS needs to be enabled first by providing a non-empty list in CorsOrigins.
	MaxAge int `mapstructure:"cmax_age"`
}
```

Reply:
I think that’d be a nice pre-1.0 cleanup, personally.

Reply:
@tigrannajaryan @codeboten @dmitryax what do you think?

Reply:
Seems like a good thing to get sorted before 1.0, a `CorsSettings` makes sense to me.

Reply:
I agree.

Reply:
@silverlyra let's do it.

Reply:
@bogdandrutu Oh you want _me_ to refactor it? 😆 Ok, I should be able to make time for that next week.

In the meantime, can someone please allow CI to run on this PR?

Reply:
Sorry @silverlyra I know I am a bad maintainer :D but want to keep the code extensible :))